### ISSUE TITLE
inherit dark mode from system theme?

### ISSUE THREAD

**Issue by BMaxV** [localsend/localsend#1459]
### Is your feature request related to a problem? Please describe.

Not really. Keeping theme consistent reduces eyestrain. But that's not really a "problem" problem.

### Describe the solution you'd like

I would like the UI to recognize that I am using dark mode on my system to switch to some dark mode, ideally the system one, but one you design would work too. It's just about the contrast.

### Additional context

I am on

OS: Ubuntu 22.04 jammy
DE: Xfce
WM: Xfwm4
WM Theme: Numix
GTK Theme: Adwaita-dark [GTK2]

---------

I don't really expect a fix, it's more of a todo, that I didn't find in your issues yet. Maybe something as a good first issue for new contributors.

Software works beautifully, thanks for making it!

**Comment by thatLeaflet** [comment]
In Settings, you can change "Color" to "System". It will then match your system theme.

**Comment by BMaxV** [comment]
This is true and I found the setting. It was already defaulting to "system". In that case something else is going on, because it "doesn't work on my machine".

![image](https://github.com/localsend/localsend/assets/5713211/f3daecaa-8dd0-4774-8d39-75d34cc2b0a9)

**Comment by thatLeaflet** [comment]
You also need to set the second option, "Color".

**Comment by BMaxV** [comment]
I got the .deb file (from the link on the github readme) and installed it via software center.

And I can confirm that setting "color" to "system" also doesn't work (as in inheriting it). 

If I set it to OLED it works, but that's because then the theme is also set to dark (which is also a bit weird). This even happens when the theme is set to "light".

![image](https://github.com/localsend/localsend/assets/5713211/6260888e-4669-4aec-b02d-0d5007245b6e)

**Comment by thatLeaflet** [comment]
Ok, so it's no sandbox issue.

My guess as to what's happening is that LocalSend uses the standardized portal for determining whether to use a light or dark appearance. But maybe XFCE isn't updating the portal value. As I understand it, XFCE doesn't have a distinct light/dark mode, it just uses a GTK theme.

Maybe try running `gsettings set org.gnome.desktop.interface color-scheme prefer-dark`

**Comment by BMaxV** [comment]
But I have *a* system theme.

What does it matter to the application whether that happens to be light or dark.

```
gsettings get org.gnome.desktop.interface color-scheme
'default'
```

So I guess that's the how. But still. It's not supposed to use a default, I have a color scheme defined.

I get that that's not your fault and I'm actually curious where and how these things work. I would really like to help and push for clarification / write code to solve this issue.

**Comment by thatLeaflet** [comment]
In the past (and into the present), many apps would try to determine whether to be light or dark based on your GTK theme. That mostly works, but it's a bit inelegant because not all desktops use GTK and so may not set a theme.

The portal method is meant to standardize things. Rather than analyzing the GTK theme, an app just asks the portal "should I use a light or dark theme" and the portal responds with a yes or no.

But XFCE is a stable but slow moving project, so  I guess they just haven't gotten around to implementing good support for the portal. Ideally XFCE should be detecting if your GTK theme is light or dark and update the portal value accordingly.

The "default" value for Gnome is a light theme, so the portal is telling LocalSend to use a light theme. Setting it to "prefer-dark" should tell LocalSend to use a dark theme instead.
